src/components/oscal-utils/OSCALRestUtils.js

```diff
@@ -184,22 +186,51 @@ export function updateSspControlImplementationImplementedRequirementStatementByC
 ) {
   const partialRestImplementedRequirement = deepClone(implementedRequirement);
 
-  // Find our statementByComponent and update the description
-  const statementByComponent = partialRestImplementedRequirement.statements
-    .find((element) => element["statement-id"] === statementId)
-    ["by-components"].find(
-      (element) => element["component-uuid"] === componentId
-    );
+  // Find our statement
+  let statement = partialRestImplementedRequirement?.statements?.find(
+    (element) => element["statement-id"] === statementId
+  );
+
+  // If our statement doesn't exist, create and add it
+  if (!statement) {
+    statement = {
+      "statement-id": props.statementId,
+      uuid: uuidv4(),
+      "by-components": [],
+    };
+    if (!partialRestImplementedRequirement.statements) {
+      partialRestImplementedRequirement.statements = [];
+    }
```

Comment:
```suggestion
    partialRestImplementedRequirement.statements ??= [];
```

This isn't really necessary but I happened to use it in some other examples so suggesting it here for consistency.

Reply:
Done.